aiida/cmdline/commands/cmd_group.py

```diff
@@ -84,6 +84,67 @@ def group_remove_nodes(group, nodes, clear, force):
         group.remove_nodes(nodes)
 
 
+@verdi_group.command('move-nodes')
+@arguments.NODES()
+@click.option('-s', '--source-group', type=types.GroupParamType(), required=True, help='The group whose nodes to move.')
+@click.option(
+    '-t', '--target-group', type=types.GroupParamType(), required=True, help='The group to which the nodes are moved.'
+)
+@options.FORCE(help='Do not ask for confirmation and skip all checks.')
+@options.ALL(help='Move all nodes from the source to the target group.')
+@with_dbenv()
+def group_move_nodes(source_group, target_group, force, nodes, all_entries):
+    """Move the specified NODES from one group to another."""
+    from aiida.orm import Group, Node, QueryBuilder
+
+    if not force and source_group.pk == target_group.pk:
+        echo.echo_critical(f'Source and target group are the same: {source_group}.')
+
+    if not nodes:
+        if not all_entries:
+            return
+        nodes = list(source_group.nodes)
+        click.confirm(f'Are you sure you want to move ALL nodes from {source_group} to {target_group}?', abort=True)
+
+    elif not force:
+
+        node_pks = [node.pk for node in nodes]
+
+        query = QueryBuilder()
+        query.append(Group, filters={'id': source_group.pk}, tag='group')
+        query.append(Node, with_group='group', filters={'id': {'in': node_pks}}, project='id')
+
+        source_group_node_pks = query.all(flat=True)
+
+        if not source_group_node_pks:
+            echo.echo_critical(f'None of the specified nodes are in {source_group}.')
+
+        if len(node_pks) > len(source_group_node_pks):
+            absent_node_pks = set(node_pks).difference(set(source_group_node_pks))
+            echo.echo_warning(f'{len(absent_node_pks)} nodes with PK {absent_node_pks} are not in {source_group}.')
+            nodes = [node for node in nodes if node.pk in source_group_node_pks]
+
+        query = QueryBuilder()
+        query.append(Group, filters={'id': target_group.pk}, tag='group')
+        query.append(Node, with_group='group', filters={'id': {'in': node_pks}}, project='id')
```

Comment:
You should probably update `node_pks` at this point to be just those pks that are actually in `source`. Now you are still using the list specified by the user, but some of which may not actually exist in `source` so the warning at line 134 would be incorrect, wouldn't it?

Edit: I see that you write explicitly in the warning ".... if present". Still wouldn't it just be clearer if you only do this for nodes that are actually in `source`?